src/Hint/NewType.hs

```diff
@@ -65,3 +74,25 @@ hasNoStrategy _                      = False
 withNewtype :: Deriving a -> Deriving a
 withNewtype (Deriving l Nothing rs)  = Deriving l (Just $ DerivNewtype l) rs
 withNewtype d                        = d
+
+-- | Given a declaration, returns the suggested \"newtype\"ized declaration following these guidelines:
+-- * @MagicHash@'d stuff is __ignored__ - @data X = X Int#@
+-- * @ExistentialQuantification@ stuff is __ignored__ - @data X = forall t. X t@
+-- * Single field constructors get newtyped - @data X = X Int@ -> @newtype X = X Int@
+-- * Single record field constructors get newtyped - @data X = X {getX :: Int}@ -> @newtype X = X {getX :: Int}@
+-- * All other declarations are ignored.
+--
+-- TODO: insert ! warning (or lack thereof) somewhere
+singleSimpleFieldNew :: Hs.HsDecl pass -> Maybe (Hs.HsDecl pass)
+singleSimpleFieldNew (Hs.TyClD wtfIsThisSomePassStuff decl@(Hs.DataDecl _ name _ _ def@(Hs.HsDataDefn _ Hs.DataType ctx _ _ [Hs.L _ constructor] derives)))
+    | simpleCons constructor = Just $ Hs.TyClD wtfIsThisSomePassStuff decl {Hs.tcdDataDefn = def {Hs.dd_ND = Hs.NewType}}
+singleSimpleFieldNew _ = Nothing
+
+-- TODO: check for MAGIC#HASH
+-- TODO: test out existentials matching
+--
+-- TODO: eventually check GADTs too
+simpleCons :: Hs.ConDecl pass -> Bool
+simpleCons (Hs.ConDeclH98 _ _ _ [] _ (Hs.PrefixCon [_]) _) = True -- TODO: does this actually check for existentials??? supposedly the first empty list holds all the existentials
```

Comment:
General suggestion is if you think of a way you might have got it wrong, but there's not a test at the top of the module that would have caught you, just add a new test case.